xarray/core/duck_array_ops.py

```diff
@@ -166,11 +126,6 @@ def cumulative_trapezoid(y, x, axis):
     return cumsum(integrand, axis=axis, skipna=False)
 
 
-masked_invalid = _dask_or_eager_func(
```

Comment:
as far as I can tell, `np.ma.masked_invalid` casts its argument to `np.ma.MaskedArray` so we might still need this (but `da.ma.masked_invalid` does not support duck arrays, either)

Reply:
It seems that dask's `masked_invalid` [is just a](https://docs.dask.org/en/latest/_modules/dask/array/ma.html#masked_invalid) `map_blocks` version of `np.ma.masked_invalid`.

> but da.ma.masked_invalid does not support duck arrays, either

It looks like this is recognised as an [upstream numpy problem](https://github.com/numpy/numpy/issues/15200), but actually has [a PR which is close to fixing it](https://github.com/numpy/numpy/pull/16022)! I'm not sure whether there is anything we can do other than wait for that PR though?

Reply:
can we just revert to the old version?

Reply:
Actually maybe we can: it looks like `masked_invalid` is only used internally.

Support for masked pint arrays would still be broken until https://github.com/numpy/numpy/pull/16022, but we don't explicitly test that in xarray.

Reply:
I've pushed a fix for this failure